src/Compilers/Core/CodeAnalysisTest/Analyzers/AnalyzerFileReferenceTests.cs

```diff
@@ -391,6 +391,77 @@ public class Generator : ISourceGenerator
             }
         }
 
+        [Fact]
+        [WorkItem(52035, "https://github.com/dotnet/roslyn/issues/52035")]
+        public void TestLoadedAnalyzerOrderIsDeterministic()
+        {
+            AnalyzerFileReference reference = CreateAnalyzerFileReference(Assembly.GetExecutingAssembly().Location);
+
+            var csharpAnalyzers = reference.GetAnalyzers(LanguageNames.CSharp).Select(a => a.GetType().FullName).ToArray();
+            Assert.Equal(4, csharpAnalyzers.Length);
```

Comment:
This test is only meaningful if the analyzers are emitted in non-alphabetical order. Think we should add a comment on the definitions to indicate that maintaining the existing order is meaningful to tests. Not a strong guard but other than inlining them here not easy to add a strong guard.

Reply:
I feel like it would be good if we could have a test resource of some kind (i.e. a helper that emits an assembly out to the temp folder which contains some generators) and test the order on that. It feels a little brittle to have to adjust these tests when a new generator is declared in the unit tests source. However, I don't feel strongly about it.

Reply:
@RikkiGibson  Yeah, I think we should probably split it up, but I think I'll defer it to the new generator work we're doing down the line, as we'll have a bunch of churn anyway.